lib/realtime/vehicle.ex

```diff
@@ -331,8 +331,9 @@ defmodule Realtime.Vehicle do
 
   def route_status(stop_id, trip, block) do
     if stop_id == List.first(trip.stop_times).stop_id do
+      first_trip_from_block = block && block |> Block.revenue_trips() |> List.first()
       # hasn't started trip yet
-      if block != nil && trip.id == List.first(block.trips).id do
+      if block != nil && trip.id == first_trip_from_block.id do
```

Comment:
I think you can get rid of the `block != nil` given the logic behind `final_trip_from_block`, right?

Reply:
that should really be `if first_trip_from_block && trip.id == first_trip_from_block.id`